src/libsystemd/sd-journal/journal-send.h

```diff
@@ -2,6 +2,23 @@
 #pragma once
 
 #include <stdbool.h>
+#include <stddef.h>
+
+#include "syslog-util.h"
 
 int journal_fd_nonblock(bool nonblock);
 void close_journal_fd(void);
+
+/* We declare sd_journal_stream_fd() as async-signal-safe. So instead of strjoin(), which calls malloc()
+ * internally, use a macro + alloca(). */
+#define journal_stream_path(log_namespace)                                              \
```

Comment:
frankly, I'd just make this a regular functin, returning heap memory.

Reply:
We can't without breaking some documented guarantees? As the comment stated.